clang/lib/Driver/ToolChain.cpp

```diff
@@ -745,7 +745,13 @@ std::string ToolChain::buildCompilerRTBasename(const llvm::opt::ArgList &Args,
   std::string ArchAndEnv;
   if (AddArch) {
     StringRef Arch = getArchNameForCompilerRTLib(*this, Args);
-    const char *Env = TT.isAndroid() ? "-android" : "";
+    const char *Env = NULL;
+    if (TT.isAndroid())
+      Env = "-android";
+    else if (TT.isOHOSFamily())
```

Comment:
This is for the old compiler-rt file hierarchy where `-android` is for compatibility. New OSes don't need this change.